### ISSUE TITLE
latest 1.8 disables new explorer file coloring when gitlens is installed

### ISSUE THREAD

**Issue by atomi** [Microsoft/vscode#38166]
- VSCode Version: 1.7
- OS Version: Mint 18

Steps to Reproduce:

1. Install GitLens
2.  coloring in file explorer is disabled

Reproduces without extensions: Yes/No
No.

https://github.com/eamodio/vscode-gitlens/issues/199

**Comment by jrieken** [comment]
@atomi I don't believe you as I use both together without issues. Please provide more information

**Comment by realbizkit** [comment]
I have installed latest v1.18 VS code and the new explorer git decorations do not work at all even after I uninstalled the GitLens extension.

I have
"git.decorations.enabled": true,
"problems.decorations.enabled": true,
"explorer.decorations.colors": true,
"explorer.decorations.badges": true,
in my user settings.

**Comment by jrieken** [comment]
Well, with those setting it is working for me and others. Please provide more information. What does the git viewlet show, what does running `git status --ignored` show? What does the git output channel show?

**Comment by cmj91uk** [comment]
@jrieken I also have issues with the latest release of VS Code (v1.18).

I previously have had gitlens installed but have since uninstalled all extensions and removed all user settings. Still no git decorations in the file explorer. I have also tried uninstalling and re-installing VSCode. `git status --ignored` shows what you would expect it to with modified files etc.

The git view shows an M next to modified files but the explorer component doesn't have color highlighting. Slightly oddly other people in the office have git color highlighting with a similar set of extensions to me!

**Comment by cmj91uk** [comment]
@realbizkit I have noticed with my local setup that the new functionality only works if you have VSCode open in the same directory as your `.git` folder but not within any of the subfolders.
Seems like a potential bug since the git view works inside of subfolders but the git decorations don't work?

**Comment by realbizkit** [comment]
**Hmm I got it working** (I had no coloring, no letters whatsoever before in the explorer view). I changed nothing. All I did is that I closed the folder and reopened it not from the homepage shortcut, but from the file menu -> Open Folder...

Before that I have restarted VS Code several times, opened my repo from the taskbar context menu shortcut etc. did not work.

**Comment by jrieken** [comment]
> I previously have had gitlens installed but have since uninstalled all extensions and removed all user settings. Still no git decorations in the file explorer.

I do not believe this has anything to do with gitlens

> I have noticed with my local setup that the new functionality only works if you have VSCode open in the same directory as your .git folder but not within any of the subfolders.

Does not reproduce for me, opening folders that subfolders of the .git-container work fine. So, please provider more information. Do you guys open a file, a folder, a workspace with a single folder, or a workspace with multiple folders? Does this reproduce with [latest insiders](https://code.visualstudio.com/insiders)?

<img width="509" alt="screen shot 2017-11-13 at 18 58 53" src="https://user-images.githubusercontent.com/1794099/32740944-da8713fc-c8a4-11e7-916e-e43667d6fd3b.png">

**Comment by vvs** [comment]
FWIW, the git coloring works fine on the latest insiders build, and I also use Git Lens, so no conflict there.
Also, I am opening the sub-folder of the git repository (this subfolder is a root of the project).

**Comment by atomi** [comment]
Yeah. This might now be related to gitlens or perhaps only slightly as I've lost coloring again and after I've uninstalled gitlens. 

Perhaps this is because I have another git repo up the folder tree that is interfering with the git status call?

**Comment by eamodio** [comment]
FYI, I'm following along on this, but I haven't seen any issues with the file explorer git coloring (and of course run GitLens). I can't see how GitLens is in anyway related, but crazy things do happen ;)

**Comment by cmj91uk** [comment]
@jrieken. I have the following setup.
Main project folder (which contains the .git directory) and a couple of subfolders down a folder for my package.
Opening from command line via "code .". Using the project switcher I can switch back to the main parent directory and all explorer coloring works as expected. Switching back to the child project doesn't restored coloring.

`git status --ignored` reports correctly (doesn't appear to be any interference from other git directories etc etc)

From a quick test of the latest insiders build everything seems to be working fine.

**Comment by jrieken** [comment]
> From a quick test of the latest insiders build everything seems to be working fine.

Hm, this might be https://github.com/Microsoft/vscode/issues/38169 where a promise will never resolve when asking for files that aren't under the current git repo...

**Comment by cleidigh** [comment]
@jrieken 
I also still think there is potentially a problem with checkIgnore using -z --stdin
I get invalid token errors when checking just two files.  Have been trying to test
and debug this for the unsaved file warning PR.  Sound possible?

**Comment by jrieken** [comment]
@cleidigh Not sure what you mean

**Comment by cleidigh** [comment]
@jrieken 
I don't have a completely clear picture yet, but I have been doing a ton of experiments trying to get something consistent to give us an example of the problem.

Basically I get an git invalid token error calling checkIgnore from my new unsaved file check
function.  My function did work with the spawn call using the command line to pass
a set of file paths to check.  When I updated and incorporated your change to use
the -z and --stdin I started to get the error.  Despite the error it looks like git works
it also exits with an exit code of zero.  the error is a catch from create_command.

I have been trying to isolate the problem but it's even weirder because I see
some inconsistencies such as when I set a local variable to a single path
within checkIgnore I can get it to succeed, if I pass into the function the same file path
I get the error.  I tried another confirmation tactic by creating a simple Python program
it does in the same command also using stdin.  I can get that working as expected.

I am going to try and create the simplest example to demonstrate the problem.  I will post a link
to my git repository when I get this, hopefully tomorrow.

**Comment by tomahl** [comment]
I've also had problems with git decorations not showing up in the explorer or editor. It worked by opening the project folder from the VS Code menu but not from within the "[Project Manager](https://github.com/alefragnani/vscode-project-manager)" extension. So I examined the "rootPath" setting, and noticed that I've used a Capital letter for the project name/folder when it should be all lowercase. After that everything worked fine. My project was located in a subdirectory from the .git folder.

**Comment by creativeworkx** [comment]
I encountered this exact behaviour today, after it worked absolutely as expected for me since .. release

Today, I installed the extension: "explorer-excluded-files" and I added _.vscode_ to files.exclude (see below) to hide it from the explorer.

```js
    "files.exclude": {
        // "**/.vscode": true
    },
```
This was causing the git decorations to be removed from explorer.
I removed _.vscode_ from files.exclude and now everything works perfectly again
